samples/azure-quantum/chemistry/Hamiltonian/Approximate Energy using Jordan-Wigner transformation.ipynb

```diff
@@ -0,0 +1,813 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "55b70453",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Preparing Q# environment...\n"
+     ]
+    }
+   ],
+   "source": [
+    "# First, import qsharp and qsharp.azure to be able to compile and submit the quantum program.\n",
+    "import qsharp\n",
+    "import qsharp.azure"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9efb9be7",
+   "metadata": {},
+   "source": [
+    "### 1. Load H2 molecule data\n",
+    "\n",
+    "The pre-generated Broombridge file included in this sample contains the details on the molecule's Hamiltonian. A Hamiltonian is an operator that calculates the energy of a molecule by acting on a qubit register that represents the molecule's quantum state by Jordan-Wigner encoding. If we prepare the molecule in the ground state, then we know that applying the Hamiltonian we will calculate the ground state energy."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "601fb1cb",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/json": "{\"LastUpdated\":\"2021-05-08T20:46:20.76547-07:00\",\"IsCompleted\":false,\"Description\":\"Adding package microsoft.quantum.chemistry.jupyter\",\"Subtask\":null}",
+      "text/plain": [
+       "Adding package microsoft.quantum.chemistry.jupyter."
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "from qdk.chemistry.broombridge import load_and_encode"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "f11f81ab",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "encoded_data_h2 = load_and_encode(\"../data/broombridge/hydrogen_0.2.yaml\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2ff4fc44",
+   "metadata": {},
+   "source": [
+    "This contains a tuple of the number of qubits, the fermionic Hamiltonian term coefficients and the energy offset.\n",
+    "\n",
+    "The index of each list in the second tuple is will be referred to as the \"term type\"."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "38c84985",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "num_qubits, fermion_terms, _, energy_offset = encoded_data_h2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "57757a37",
+   "metadata": {},
+   "source": [
+    "Reload the workspace to load the Q# program for import into Python."
```

Comment:
Last time I checked we can't use `Microsoft.Quantum.Chemistry` because of the hardware limitations.

Reply:
I thought there were a few specific operations that couldn't be used? Regardless, `import qsharp.chemistry` causes the `Microsoft.Quantum.Chemistry` NuGet package to be loaded (hence why the `qsharp.reload()` is needed to inject that into the local workspace). Adding that package directly to the project file allows IQ# to know that you need Microsoft.Quantum.Chemistry from startup rather than compiling, failing, adding the package, and then reloading.

Reply:
I don't think we can include `Microsoft.Quantum.Chemistry` to a sample that is submitted to the Azure Quantum service. This is the reason why they are in two separate folders.

Reply:
The reason why I added `qsharp.reload()` is that if there is a compilation error in the Q#, the import silently fails. `qsharp.reload()` triggers a compilation which then displays the error.